src/Weasel.Postgresql/Tables/NameExtensions.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace Weasel.Postgresql.Tables
+{
+    public static class NameExtensions
+    {
+        public const int NAMEDATALEN = 64; // Postgres default for name limits (src/include/pg_config_manual.h)
```

Comment:
It would be better to use run a query on db `SHOW max_identifier_length` to get the max identifier length for checks. Constant value of `64` bytes is good as a default but if someone updated this value and did a custom build of Postgres, our assumptions of this default value will turnout to be incorrect.

Reply:
Yes, I agree, although I'm pretty sure most people don't run a custom build of Postgres and this at least fixes the bug for the vast majority of people.

To be honest, it is beyond my knowledge of the Weasel architecture to implement that query, so I'd need a little bit of help:  Where would I run this query on the db and where do I store that value, such that it can be used in the `TableDelta` class when checking the names?

Reply:
Agree that most of the cases will use the default. Let me check and provide you some pointers on how to implement the suggestion.

Reply:
The following are the pointers for implementation:
- Add the following to `public partial class Table : ISchemaObject`

```cs
/// <summary>
/// Max identifier length for identifiers like table name, column name, constraints, primary key etc
/// </summary>
public int MaxIdentifierLength { get; set; } = 63;
```
Note that the max value is 63, this is the value which I query with `SHOW max_identifier_length`

- Update `Table.ConfigureQueryCommand(...)` to fetch the `max_identifier_length`

```cs
....
....
order by column_index;

SHOW max_identifier_length;
```

- Add a new method `readMaxIdentifierLength(...)`

```cs
private static async Task readMaxIdentifierLength(DbDataReader reader, Table existing)
{
    await reader.NextResultAsync();
    
    while (await reader.ReadAsync())
    {
        existing.MaxIdentifierLength = int.Parse(await reader.GetFieldValueAsync<string>(0));
    }
}
```

- Call method `readMaxIdentifierLength(...)` in `readExisting(DbDataReader reader) method`

```cs
....
....
await readMaxIdentifierLength(reader, existing);
            
return !existing.Columns.Any() 
    ? null 
    : existing;
```

- Now you can adjust your existing implementation to use property `MaxIdentifierLength`

Reply:
Thanks, this was already the complete solution of course :-)

I've updated the PR

Reply:
Blech, very happy that you guys sorted that out before I was aware of it. Thumbs up from me.